status/controller_test.go

```diff
@@ -33,7 +33,21 @@ var _ = Describe("Controller", func() {
 		controller = status.NewController[*TestObject](kubeClient, recorder)
 		ctx = log.IntoContext(context.Background(), ginkgo.GinkgoLogr)
 	})
+	It("should emit termination metrics when deletion timestamp is set", func() {
```

Comment:
This doesn't test that we are actually emitting the metric, right?

Reply:
It does. Added more checks in the test to validate duration seconds in greater than 0.